Added loading wheel and fixed DOM elements

Lucytheanimefan opened the pull request:
Enabled drilling down on graph, added css styling, and incorporated other more recent updates such as clickable x-axis labels

Lucytheanimefan committed:
Merge remote-tracking branch 'refs/remotes/phenoscape/master'

Lucytheanimefan committed:
Updated taxa visual

Fixed taxa integration so that drilling down works

Lucytheanimefan committed:
Fixed styling for graph

Added graph.css

hlapp commented:
@balhoff feel free to use this as an opportunity to provide code review comments.

Lucytheanimefan committed:
Added both graphs to skeleton

Lucytheanimefan committed:
Revert "Added both graphs to skeleton"

This reverts commit 30ed0cd16ebb27d18e26e3fe8715fe437e09a4ea.

Lucytheanimefan committed:
Revert "Revert "Added both graphs to skeleton""

This reverts commit b978ea6df21aa5ab3cbfe6deb764c3b1d8f0b41c.

Lucytheanimefan committed:
Fix Changes

Lucytheanimefan committed:
Added both graphs

balhoff commented:
@Lucytheanimefan looking nice! The most immediate piece of feedback I have is wondering whether you can draw the graphs within a particular DOM element. Right now they are being appended to the end of the page after the footer. The function where the drawing is implemented gets 3 arguments: `link: function(scope, element, attrs)`. All the content for this directive should go inside of `element`. Is that an easy fix?

balhoff commented on js/directives.js:
```diff
@@ -4,434 +4,760 @@
+            return {
+                restrict: 'E',
+                replace: false,
+                link: function(scope, element, attrs) {
```
Here the element is passed within which to do the drawing.

balhoff commented on index.html:
```diff
@@ -74,6 +75,7 @@
                 </div>
             </div>
         </div>
+        <script src="http://labratrevenge.com/d3-tip/javascripts/d3.tip.v0.6.3.js"></script>
```
Remove this line. d3.tip is included below. This one causes a JavaScript error because d3 hasn't been included yet.

balhoff commented on js/directives.js:
```diff
@@ -4,434 +4,760 @@
+                                    d3.select(this).style("fill", "black");
+                                })
+                                .on("click", function(d) {
+                                    document.location.href = "http://kb.phenoscape.org/#/entity/" + data[d][1];
```
This line causes a JavaScript error when you click on a taxon name in the X axis of the taxa graph. If you first drill down one level, it works without error.

balhoff commented:
It would be nice to add a scope variable where you could set `true` or `false` if the graph is busy retrieving data. The app could watch this variable and put up a busy indicator (you don't need to put this functionality in your directive).

Lucytheanimefan committed:
Deleted src to labratrevenge

Lucytheanimefan committed:
Fixed hyperlinks for taxa graph x axis label

X axis labels for the base case now link to the correct corresponding
taxa page on the KB

Lucytheanimefan committed:
Added spinning loading wheel

Lucytheanimefan committed:
Added all the elements as children of the ‘element’ argument

Lucytheanimefan committed:
Repositioned loading wheel

hlapp commented:
@balhoff can you comment whether you consider this ready for merge now?

balhoff commented:
@Lucytheanimefan nice fixes! The graphs are placed where I expect now.
